gitpod-io/gitpod
# [db] Move migrations from *.sql to TS

## Description
Fixes the approach introduced with https://github.com/gitpod-io/gitpod/pull/6541

## Related Issue(s)

Fixes #

## How to test
 - open workspace on this branch
 - go to `components/gitpod-db` and run `yarn watch`
 - open new terminal:
 - `kubectl port-forward mysql-0 23306:3306`
 - open a new terminal: check the current schema (should be the "new" format, incl. `id` column):
   -  `mysql -u gitpod -h 127.0.0.1 -P 23306 -p`
   - `describe migrations;`
   - `select * from migrations` and see how id is sorted by timestamp
 - open `migrate-migrations.ts`, switch `up(...)` to `down(...)`
 - open new terminal, `cd components/gitpod-db` and run:
   - `yarn run migrate-migrations`
   - now how the table is now back to the old format (`id` column gone, `timestamp` is primary key)
 - open `migrate-migrations.ts` and revert your change
 - `yarn run migrate-migrations`:
   - `describe migrations;`
   - `select * from migrations` and see how id is sorted by timestamp
 - run `yarn run migrate-migrations` again and note how it's idempotent
 
## Release Notes

```release-note
NONE
```

## Documentation
